src/OpenTelemetry.Api/Trace/SpanAttributes.cs

```diff
@@ -130,6 +130,16 @@ public void Add(string key, double[] values)
         this.AddInternal(key, values);
     }
 
+    /// <summary>
+    /// Add entry to the attributes.
+    /// </summary>
+    /// <param name="key">Entry key.</param>
+    /// <param name="value">Entry value.</param>
+    public void Add(string key, object value)
```

Comment:
what is the use case for this? Though Activity allow `object` type, OTel only allows a specific set of types here, which is already supported in this class.

Reply:
In this case, it was to be able to have an object rendered into a string out of band with the hot path being traced. I'm on mobile now so I can't provide a code sample, but you can already accomplish this by passing an IEnumerable of KV<string,object> into the constructor.

Exposing this method eliminates the need to do that and writes to the underlying collection.

Reply:
Having looked at the spec, it doesn't preclude SDKs from having this method, just that the others MUST exist. I don't see a reason that the .NET SDK can't have extra stuff that isn't in the base spec?

Reply:
I don't think we should be adding this overload for the following reasons:

1. You don't need a new API to defer the `ToString()` call. You should be able to use the existing `SpanAttributes` ctor that takes in an `IEnumerable<KeyValuePair<string, object>>` to directly pass in a complex type. Could you talk more about why this option wouldn't work for you?
2. Just because .NET's `Activity` API deviates from the spec by allowing an arbitrary object as an attribute value, doesn't mean we should give up on keeping the `Span` and related APIs conform to the spec.